packages/core/src/routes/session/username-password.ts

```diff
@@ -109,4 +110,35 @@ export default function usernamePasswordRoutes<T extends AnonymousRouter>(
       return next();
     }
   );
+
+  router.post(
+    reAuthRoute,
+    koaGuard({
+      body: object({
+        password: string().min(1),
+      }),
+    }),
+    async (ctx, next) => {
+      const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+      assertThat(
+        result?.login?.accountId,
+        new RequestError({
+          code: 'auth.unauthorized',
+          status: 401,
+        })
+      );
+
+      const user = await findUserById(result.login.accountId);
+      assertThat(user.username, 'session.invalid_sign_in_method');
+
+      const { password } = ctx.guard.body;
+      const type = 'ReAuthUsernamePassword';
+      ctx.log(type, { username: user.username, userId: user.id });
```

Comment:
nit: shall we build a literal enum variable to index log types instead of typing a string?